### ISSUE TITLE
delegate_traps assertion failed when spike supports H extension

### ISSUE THREAD

**Issue by syvenwang7** [riscv/riscv-pk#214]
When spike supports H extension, the some bits of  mideleg are tied to 1.
The assertions in delegate_traps function are failed.

**Comment by aswaterman** [comment]
I think pk needs to clear the misa.H bit.

cc @avpatel - in Spike, does clearing the misa.H bit get rid of the mideleg bits that are tied to 1

**Comment by avpatel** [comment]
Spike does not clear the VS-interrupt bits tied to 1 in mideleg when misa.H bit is updated. We have to fix Spike for this.

Regards,
Anup

**Comment by aswaterman** [comment]
Yeah, I think that is a minor bug that needs to get fixed.  In the mean time, you can work around the bug by commenting out the assertion, I hope.  (But we shouldn't upstream the workaround.)

**Comment by avpatel** [comment]
I have send PR to Spike repo. Link https://github.com/riscv/riscv-isa-sim/pull/537/
